backend/src/main/java/com/capstone/moneytree/controller/StockController.java

```diff
@@ -0,0 +1,83 @@
+package com.capstone.moneytree.controller;
+
+import com.capstone.moneytree.service.api.StockMarketDataService;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.*;
+
+import com.capstone.moneytree.service.api.StockService;
+import pl.zankowski.iextrading4j.api.stocks.*;
+import pl.zankowski.iextrading4j.api.stocks.v1.BatchStocks;
+import pl.zankowski.iextrading4j.api.stocks.v1.KeyStats;
+import pl.zankowski.iextrading4j.api.stocks.v1.News;
+
+import javax.validation.Valid;
+import javax.validation.constraints.NotBlank;
+import javax.validation.constraints.Size;
+import java.util.List;
+
+@RestController
+@RequestMapping("/api/stockmarket")
+public class StockController extends ApiController {
+
+   @Autowired
+   private StockMarketDataService stockMarketDataService;
+   private static final Logger LOG = LoggerFactory.getLogger(StockController.class);
+
+   /**
+    * Obtain json object containing company quote, info, news, keystats and logo
+    *
+    * @param symbol
+    * @return batchStocks
+    */
+   @GetMapping("/batch/{symbol}")
+   public ResponseEntity<BatchStocks> getBatchStocksBySymbol(@Valid @PathVariable(name = "symbol") String symbol) {
+      BatchStocks batchStocks = stockMarketDataService.getBatchStocksBySymbol(symbol);
+      return ResponseEntity.ok(batchStocks);
```

Comment:
Do you think we should return ResponseEntity.of(Optional response) here so that we can handle good and bad requests later on? Because right now  we assume that all requests made are of response 200 if I understood correctoy. It's fine for now though.

Reply:
We can handle this next sprint. A story was made for this #122.